src/sql/util.py

```diff
@@ -363,3 +369,22 @@ def show_deprecation_warning():
         "raise an exception in the next major release so please remove it.",
         FutureWarning,
     )
+
+
+def is_saved_snippet_or_table_exists(table, schema=None):
+    """
+    Check if the referenced table is a snippet
```

Comment:
has this moved from `magic_plot.py`. then it should be deleted from `magic_plot.py` and `util. is_saved_snippet_or_table_exists()` to be called instead